crates/wasmtime/src/module.rs

```diff
@@ -300,11 +301,21 @@ impl Module {
     }
 
     unsafe fn compile(engine: &Engine, binary: &[u8]) -> Result<Self> {
-        let compiled = CompiledModule::new(
-            engine.compiler(),
-            binary,
+        #[derive(Hash)]
+        struct Env<'a>(&'a Compiler, &'a [u8]);
+
+        let cache_entry = ModuleCacheEntry::new("wasmtime", engine.cache_config());
+        let artifacts = cache_entry.get_data(
+            Env(engine.compiler(), binary),
+            |Env(compiler, binary)| -> Result<_> {
+                Ok(CompilationArtifacts::build(compiler, binary)?)
+            },
```

Comment:
Can this pass `CompilationArtifacts::build` as a bare function instead of making a closure?

Reply:
remove `Env`, but I do not see how to unwrap tuple to pass to CompilationArtifacts::build as params

Reply:
Oops, indeed!